metaserver/src/main/scala/scala/meta/languageserver/Configuration.scala

```diff
@@ -1,12 +1,73 @@
 package scala.meta.languageserver
 
 import play.api.libs.json._
+import play.api.libs.functional.syntax._
+
+import org.langmeta.AbsolutePath
+import org.langmeta.RelativePath
+import java.nio.file.Paths
+
+import Configuration._
+import play.api.libs.json.OFormat
+import scala.meta.languageserver.Configuration
+import scala.meta.languageserver.Configuration.{ Completions, Experimental, Hover, Indexing, Rename, Scalafix, Scalafmt }
 
 case class Configuration(
-    scalafmtConfPath: String,
-    scalafixConfPath: String,
-    enableCompletions: Boolean,
+  scalafmt: Scalafmt = Scalafmt(),
+  scalafix: Scalafix = Scalafix(),
+  indexing: Indexing = Indexing(),
+  experimental: Experimental = Experimental()
 )
+
 object Configuration {
+
+  case class Experimental(
+      completions: Completions = Completions(),
+      hover: Hover = Hover(),
```

Comment:
I think we can move hover and rename out of Experimental since both are built on semanticdb, esp. with edit distance we have quite a good story to reduce bogus results. The only part I'm not convinced about enabling by default is building semanticbs on the fly with the presentation compiler and completions.